### PR TITLE
Ensure Client is safe for concurrent use

### LINKED ISSUE #6287: Influx client has a race
There is unprotected access to fields on the client's `net.URL` field. This results in a data race when the client is concurrently used to query, write or ping.

### PR DESCRIPTION
**Pull request by e-dard**
###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated

Fixes #6287.

### CODE REVIEW

File: client/v2/client_test.go [hunk 1]
```diff
@@ -138,6 +140,54 @@ func TestClient_Ping(t *testing.T) {
 	}
 }
 
+func TestClient_Concurrent_Use(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		fmt.Fprintf(w, `{}`)
```
Comment (jsternberg) [inline client/v2/client_test.go]: You can just use `io.WriteString(w, `{}`)` here.

**Comment by jsternberg** [comment]
:+1: with one minor nitpick.

File: client/v2/client_test.go (hunk 1 above)
Comment (e-dard) [inline client/v2/client_test.go]: or better yet `w.Write` then I don't need to import `fmt` or `io` at all 🎉

**Comment by corylanou** [comment]
I've been bit by that URL so many times in the past... +1

**Pull request merged**